contextual_orchestrator/provider_catalog.py

```
@@ -0,0 +1,271 @@
+    if not api_key:
+        return []
+    parsed = urlparse(base_url)
+    if not parsed.hostname:
+        return []
+    if not allow_insecure and parsed.scheme != "https":
+        return []
+    request = Request(
+        f"{base_url.rstrip('/')}/models",
+        headers={"authorization": f"Bearer {api_key}", "accept": "application/json"},
+        method="GET",
+    )
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)